### PR TITLE
Add additional Ethereum block types: FullBlock, FullBlockWithReceipts

### PR DESCRIPTION
**Pull request by fordN**
This PR contributes to the new feature that allows subgraph developers to choose between different Ethereum block structs to use as the input argument to a block handler mapping function. 

New Ethereum block types are included in the Ethereum module: `FullBlock`, `FullBlockWithReceipts`, and `TransactionWithReceipt`

### CODE REVIEW

File: chain/ethereum.ts [hunk 1]
```diff
@@ -352,6 +352,48 @@ export namespace ethereum {
     size: BigInt | null
   }
 
+  /**
+   * An Ethereum block with transactions.
+   */
+  export class FullBlock {
```
Comment (That3Percent) [inline chain/ethereum.ts:358]: Why not less boilerplate with:

```typescript
export class FullBlock extends Block {
   transactions: Array<Transaction>
}
```

The same trick applies to the other classes that have been introduced. I'm a fan of using interfaces for consistency personally, but don't feel the urge to evangelize it too much today. Here's what that would look like (if I remember correctly, my TS is rusty... lol):
```typescript
export type FullBlock = Block & {
   transactions: Array<Transaction>,
}
```
One nice thing about interfaces here is that if `TransactionWithReceipt` implements `Transaction` then `FullBlockWithReceipts` also implements `FullBlock`. Type/Interface compatibility isn't a huge concern for this use-case, but it's interesting to discuss.

**Comment by zinootje** [comment]
the logs parameter (`logs: Array - Array of log objects, which this transaction generated ` ) wich is part of the transaction receipt  missing from the TransactionWithReceipt type . Could you maybe consider adding this ?

**Comment by leoyvens** [comment]
Hi @zinootje, because logs are the largest part of a receipt, for performance and operating cost reasons they are deliberately left out here. We're designing this so that it is possible for us to add the option of retrieving the receipts with logs. What is your use case for this?

**Comment by zinootje** [comment]
@leoyvens . I am trying to make a subgraph on a  dex aggregator. I watch for the event the dex aggregator contract emits on a swap and then i would parse  logs of the transaction to see which protocols are used an how. For example if the Dex agregator uses uniswap there will be a event emitted by the uniswappool contract , if it uses bancor a event emited by the bancor poolcontract , ... .

**Comment by franz101** [comment]
Whats the status here?

**Comment by azf20** [comment]
Hey @zinootje your request is supported now in https://github.com/graphprotocol/graph-node/pull/3373 (thanks @tilacog )
@franz101 we are not working to add receipts to blocks right now, do you have a specific use case in mind?

**Comment by leoyvens** [comment]
See the documentation for this feature here https://thegraph.com/docs/en/developer/create-subgraph-hosted/#transaction-receipts-in-event-handlers.

**Pull request closed without merging**